### PR TITLE
Refactor geotiff export test to work with gdal1/2 and python2/3.

### PR DESCRIPTION
**Pull request by pp-mo**
his should fix test failures of iris.tests.experimental.test_raster, which is currently failing in the Python3 test when "unpinned".

N.B. this is based against master, so does not test changes, except to show it still works in the old "pinned" context

### CODE REVIEW

File: lib/iris/tests/experimental/test_raster.py [hunk 1]
```diff
@@ -29,29 +29,42 @@
...
-                                for tag, val in sorted(im.tag.items()))
-        reference_path = tests.get_result_path(reference_filename)
-        self.assertString(tiff_header, reference_path)
-
-    def check_tiff(self, cube, tif_header):
+        with open(tiff_filename, 'rb') as fh:
+            im = PIL.Image.open(fh)
+            file_keys = im.tag.keys()
+
+            missing_keys = sorted(list(set(expect_keys) - set(file_keys)))
```
Comment (QuLogic) [inline lib/iris/tests/experimental/test_raster.py]: The `list` is unnecessary.

File: lib/iris/tests/experimental/test_raster.py [hunk 2]
```diff
@@ -29,29 +29,42 @@
...
+            msg_nokeys = "Tiff header has missing keys : {}."
+            self.assertEqual(missing_keys, [],
+                             msg_nokeys.format(missing_keys))
+
+            extra_keys = sorted(list(set(file_keys) - set(expect_keys)))
```
Comment (QuLogic) [inline lib/iris/tests/experimental/test_raster.py]: The `list` is unnecessary.

**Comment by marqh** [comment]
see https://github.com/marqh/iris/pull/29 for unpinned testing (a bit noisy, but supportive or this PR)

**Comment by marqh** [comment]
obsoleted by #2188, closing

**Pull request merged**